src/model/neural_network/neural_network.py

```diff
@@ -0,0 +1,104 @@
+from typing import List
+
+import numpy as np
+
+from model.neural_network.activation_function_utils import (
+    activation_functions_utils,
+    ActivationFunctionUtils,
+)
+
+np.set_printoptions(suppress=True)
+
+
+class NeuralNetworkHiddenLayerInfo:
+    def __init__(self, activation_function_name: str, neurons_count: int):
+        self.activation_function_name: str = activation_function_name
+        self.neurons_count: int = neurons_count
+
+
+class NeuralNetworkHiddenLayer:
+    def __init__(self, basic_info: NeuralNetworkHiddenLayerInfo, weights: np.ndarray, biases: np.ndarray):
+        self.info = basic_info
+        self.utils: ActivationFunctionUtils = activation_functions_utils[
+            basic_info.activation_function_name
+        ]
+        self.weights: np.ndarray = weights
+        self.biases = biases
+        self.values: np.ndarray = np.zeros(shape=(self.info.neurons_count,))
+
+
+class NeuralNetwork:
+    """
+    Class representing a neural network (multilayer perceptron).
+    """
+    def __init__(
+            self,
+            hidden_layers_info: List[NeuralNetworkHiddenLayerInfo],
+            output_neurons_count
+    ):
+        """
+        Args:
+            hidden_layers_info (List[NeuralNetworkHiddenLayerInfo): A list of objects
+                with info needed for creation of hidden layers. There must be at least two layers,
+                since single layer network is not supported.
+        """
+        if len(hidden_layers_info) < 2:
+            raise NotImplementedError(
+                "This implementation of neural network does not support single layer networks"
+            )
+        self.output_neurons_count = output_neurons_count
+        self.output = np.zeros((output_neurons_count, 1))
+
+        self.hidden_layers_info = hidden_layers_info
+        self.input_layer_neuron_count = self.hidden_layers_info[0].neurons_count
+        self.hidden_layers = self._create_hidden_layers()
+
+        self.eta = 0.5
+
+    def _feed_forward(
+            self,
+            training_data_sets: np.ndarray,
+    ) -> None:
+        """
+        Feeds the neural network with the data provided.
+
+        Args:
+            training_data_sets (numpy.ndarray): Data which will be used for feeding the network.
+        """
+        input_layer = self.hidden_layers[0]
+        input_layer.values = training_data_sets
+        for layer, next_layer in zip(self.hidden_layers[0:-1], self.hidden_layers[1:]):
```

Comment:
TODO: replace `zip` with `utils.pairwise` after it's merged.

Reply:
Not needed anymore due to simplification made